### PR TITLE
Determine whether a given sanitizer is available based on the presenc…

### PR DESCRIPTION
**Pull request by cheshire**
…e of the library.

rdar://37192887

Basically, if the library is there, the sanitizer is available. IF not, then it's not available.

### CODE REVIEW

**Comment by cheshire** [comment]
@swift-ci please test

File: lib/Option/SanitizerOptions.cpp [hunk 1]
```diff
@@ -108,33 +120,34 @@ llvm::SanitizerCoverageOptions swift::parseSanitizerCoverageArgValue(
...
+        .Case("address", SanitizerKind::Address)
+        .Case("thread", SanitizerKind::Thread)
+        .Case("fuzzer", SanitizerKind::Fuzzer)
+        .Default(None);
+    bool isShared = kind && *kind != SanitizerKind::Fuzzer;
+    if (!kind) {
       Diags.diagnose(SourceLoc(), diag::error_unsupported_option_argument,
           A->getOption().getPrefixedName(), A->getValue(i));
+    } else if (sanitizerRuntimeLibExists(toFileName(*kind), isShared)
+        && (*kind != SanitizerKind::Thread || Triple.isArch64Bit())) {
```
Comment (kubamracek) [inline lib/Option/SanitizerOptions.cpp]: Can we keep the helper function "isTSanSupported"? Even if it's just a test for .isArch64bit. This condition is very hard to read.

**Comment by kubamracek** [comment]
Besides the comment, this LGTM. Thanks!

File: lib/Option/SanitizerOptions.cpp (hunk 1 above)
Comment (cheshire) [reply lib/Option/SanitizerOptions.cpp]: @kubamracek keeping the function does not make sense, as we now check existence of all sanitizers. I've tried to rewrite the condition to make it more readable.

Comment (kubamracek) [reply lib/Option/SanitizerOptions.cpp]: Can we then just be a bit more verbose here? Something like
```
bool sanitizerSupported = true;
if (*kind == Thread) {
  if (!Triple.is64bit() sanitizerSupported = false; // TSan requires 64-bit architecture
}
...
```

**Comment by cheshire** [comment]
@swift-ci please smoke test

**Comment by kubamracek** [comment]
LGTM. Thanks!

**Review by kubamracek — APPROVED** [review]

**Pull request merged**